### ISSUE TITLE
API mistake in transaction

### ISSUE THREAD

**Issue by ecfish** [cmu-db/bustub#142]
Hi~ I am working on B+ tree index. I want to insert index write record into transaction's index_write_set_, but mistake may exist here:
https://github.com/cmu-db/bustub/blob/master/src/include/concurrency/transaction.h#L200
Should this be AppendIndexWriteRecord(...)?

Thanks~

**Comment by apavlo** [comment]
Yes I think that you are correct. Can you send a PR with the fix?

**Comment by ecfish** [comment]
#143 
Done!

**Comment by lmwnshn** [comment]
Thanks for the PR, sorry for the delay!
